provisions/roles/db/templates/cron_cccp_db_read_check.sh.j2

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+INPUT_DIR="{{ db_backup_nfs_path }}"
+
+DBUSER="{{ db_user }}"
+DBNAME="{{ db_name }}"
+DBPASS="{{ db_pass }}"
+
+PGIMAGE="{{ postgresql_image }}"
+
+# find latest tar file in $INPUT_DIR first
+LATEST=$(ls -Art $INPUT_DIR | grep cccp-pgdump.tar | tail -n 1)
+# find its absolute path
+LATEST_DUMP=$(find $INPUT_DIR -name $LATEST)
+
+PGSERVERNAME="postgresql_restore"
+
+function exit_on_non_zero
+{
+  if [[ $1 != 0]]; then
+    echo $2
+    # stop and remove the restore container
+    echo "Removing $PGSERVERNAME container"
+    docker stop $PGSERVERNAME && docker rm $PGSERVERNAME
+    exit $1
+  fi
+}
+
+echo "==================================================================="
+echo "`date +'%F %T'` Starting restore test for  $LATEST_DUMP"
+echo "-------------------------------------------------------------------"
+echo "File to be tested is $LATEST_DUMP"
+
+# run the server
+echo "Running the postgresql server.."
+docker run -d -v $INPUT_DIR:$INPUT_DIR --name $PGSERVERNAME -e "POSTGRESQL_DATABASE=$DBNAME" -e "POSTGRESQL_USER=$DBUSER" -e "POSTGRESQL_PASSWORD=$DBPASS" $PGIMAGE
+exit_on_non_zero $? "Failed to run postgresql server for restore of $LATEST_DUMP."
+echo "Wait for server to start up.."
```

Comment:
I am assuming we are restoring to a temporary postgresql container, instead of main one, because we dont want to disrupt that?

Reply:
yes, that's correct.

Reply:
Thanks